mlir-tensorrt/tensorrt/lib/TensorRT/Transforms/RaiseNormalizations.pdll

```diff
@@ -219,6 +223,83 @@ Constraint Mean(input: Value, numHW: Value){
  return Div(ExpandTailDims(ReduceSum(FlattenTailDims(input))), numHW);
 }
 
+Constraint ReduceAvg(input: Value, reduceAxes: Attr) {
+    let avgOp = op<tensorrt.reduce>(input) {keepDimensions = attr<"true">, reduceAxes = reduceAxes};
+    AvgImpl(avgOp);
+    return avgOp;
+}
+
+
+Rewrite GetSplatElementAttr(x: Value) -> Attr [{
+  while(true) {
+    if(auto expandRank = x.getDefiningOp<tensorrt::ExpandRankOp>())
+      x = expandRank.getInput();
+    else if(auto reshape = x.getDefiningOp<tensorrt::ReshapeOp>())
+      x = reshape.getInput();
+    else if(auto broadcast = x.getDefiningOp<tensorrt::BroadcastOp>())
+      x = broadcast.getInput();
+    else if(auto cast = x.getDefiningOp<tensorrt::CastOp>())
+      x = cast.getInput();
+    else if(auto identity = x.getDefiningOp<tensorrt::IdentityOp>())
+      x = identity.getInput();
+    else if(auto slice = x.getDefiningOp<tensorrt::SliceOp>())
+      x = slice.getInput();
+    else if(auto constant = x.getDefiningOp<tensorrt::ConstantOp>()) {
+      DenseElementsAttr els{};
```

Comment:
Nit: you can eliminate lines 251-252 by declaring `els` as `SplatElementsAttr`.

Reply:
done